so3/fs/vfs.c

```diff
@@ -824,6 +838,59 @@ SYSCALL_DEFINE3(fcntl, int, fd, unsigned long, cmd, unsigned long, args)
 	return 0;
 }
 
+/*
+ * Implementation of the writev syscall
+ */
+SYSCALL_DEFINE3(writev, unsigned long, fd, const struct iovec *, vec, unsigned long, vlen)
+{
+	int i;
+	int ret;
```

Comment:
```suggestion
	int ret = 0;
```

So we don't send garbage if `vlen == 0`